astropy/coordinates/sky_coordinate.py

```diff
@@ -1862,15 +1862,15 @@ def guess_from_table(cls, table, **coord_kwargs):
 
         Parameters
         ----------
-        table : astropy.Table
+        table : `~astropy.table.Table` or subclass
```

Comment:
Is there a table-like and is that relevant here?

Reply:
There *is* a table-like. I don't think it's relevant, because table-like includes dictionaries and anything that can initialize a Table. In general, a parameter in a function is "-like" only if it is wrapped with the constructor. In this PR I refrain from doing that. I think it would be a very nice (large) set of followup PRs to expand input options by trivially wrapping inputs with the relevant constructor (and ``copy=False``, if present).